src/Arcus.Security.Providers.HashiCorp/Extensions/SecretStoreBuilderExtensions.cs

```diff
@@ -48,8 +81,49 @@ public static class SecretStoreBuilderExtensions
+            Guard.NotNullOrWhitespace(username, nameof(username), "Requires a username for the UserPass authentication during connecting with the HashiCorp Vault");
+            Guard.NotNullOrWhitespace(password, nameof(password), "Requires a password for the UserPass authentication during connecting with the HashiCorp Vault");
+            Guard.NotNullOrWhitespace(secretPath, nameof(secretPath), "Requires a path where the HashiCorp Vault secrets are stored");
+            Guard.For<ArgumentException>(() => !Uri.IsWellFormedUriString(vaultServerUriWithPort, UriKind.RelativeOrAbsolute), "Requires a HashiCorp Vault server URI with HTTP port");
+
+            return AddHashiCorpVaultWithUserPass(builder, vaultServerUriWithPort, username, password, secretPath, configureOptions, name: null, mutateSecretName: null);
+        }
+
+        /// <summary>
+        /// <para>
+        ///     Adds the secrets of a HashiCorp Vault KeyValue engine to the secret store.
+        /// </para>
+        /// <para>
+        ///     See more information on HashiCorp: https://www.vaultproject.io/docs.
+        /// </para>
+        /// </summary>
+        /// <param name="builder">The builder to add the HashiCorp secrets to.</param>
+        /// <param name="vaultServerUriWithPort">The URI that points to the running HashiCorp Vault.</param>
+        /// <param name="username">The username of the UserPass authentication method.</param>
+        /// <param name="password">The password of the UserPass authentication method.</param>
+        /// <param name="secretPath">The secret path where the secret provider should look for secrets.</param>
+        /// <param name="configureOptions">The optional function to set the additional options to configure the HashiCorp Vault KeyValue.</param>
+        /// <param name="name">The unique name to register this HashiCorp provider in the secret store.</param>
+        /// <param name="mutateSecretName">The optional function to mutate the secret name before looking it up.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> or <paramref name="secretPath"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="vaultServerUriWithPort"/> is blank or doesn't represent a valid URI,
+        ///     or the <paramref name="username"/> or <paramref name="password"/> is blank,
+        ///     or the <paramref name="secretPath"/> is blank.
+        /// </exception>
+        public static SecretStoreBuilder AddHashiCorpVaultWithUserPass(
+            this SecretStoreBuilder builder,
+            string vaultServerUriWithPort,
+            string username,
+            string password,
+            string secretPath,
+            Action<HashiCorpVaultUserPassOptions> configureOptions,
+            string name,
```

Comment:
General remark, can't we make it the first parameter?

This is also how https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks does it

Reply:
The name parameter is also the first one after the specific parameters of the secret provider itself. Same as here. Or am I missing something?

Reply:
It's ok!